src/reducers/index.js

```diff
@@ -287,19 +291,26 @@ const getTxHistoryFromWSTx = (tx, tokenUid, tokenTxBalance) => {
 
 /**
  * Got wallet history. Update wallet data on redux
+ * @param {string[]} action.payload.tokens
+ * @param {string} action.payload.currentAddress
+ * @param {{uid:string, name:string, symbol:string}[]} action.payload.registeredTokens
  */
 const onLoadWalletSuccess = (state, action) => {
   // Update the version of the wallet that the data was loaded
   LOCAL_STORE.setWalletVersion(VERSION);
   const { tokens, registeredTokens, currentAddress } = action.payload;
-  const allTokens = new Set(tokens);
 
+  // Convert the tokens list to a new Set to ensure there are no duplicated tokens
+  const allTokens = new Set();
+  for (const key in tokens) {
+    allTokens.add(tokens[key]);
+  }
   return {
     ...state,
     loadingAddresses: false,
     lastSharedAddress: currentAddress.address,
     lastSharedIndex: currentAddress.index,
-    allTokens,
+    allTokens: helpersUtils.convertSetToObject(allTokens),
```

Comment:
Do we need this?

If the `allTokens` object will have key and value as the uid, it means that it already will filter out possible duplicates.
We could just have:

```javascript
const allTokens = {};
for (const key in tokens) {
  allTokens[key] = key;
}
```

And the allTokens below would work the same way, right?

Reply:
That's a good approach. I implemented it on https://github.com/HathorNetwork/hathor-wallet/pull/587/commits/3ee5ac5cfba3a1e7dbe5387109159d06d4a34485